src/server/bitops_family.cc

```diff
@@ -343,8 +343,14 @@ std::string ElementAccess::Value() const {
 
 void ElementAccess::Commit(std::string_view new_value) const {
   if (shard_) {
-    element_iter_->second.SetString(new_value);
-    post_updater_.Run();
+    if (new_value.empty()) {
+      // No need to run, it was a new entry and it got removed
+      post_updater_.Cancel();
```

Comment:
I think you should run the post updater here as this might be not a new entry

Reply:
yep +1. I wish we could run it unconditionally though (but we can't because we have an assertion within it :(